peer/pendingheap/heap.go

```diff
@@ -90,8 +89,7 @@ func (ph *pendingHeap) Remove(p peer.StatusPeer, _ peer.Identifier, sub peer.Sub
 	ph.Unlock()
 }
 
-func (ph *pendingHeap) notifyStatusChanged(ps *peerScore) {
-	status := ps.peer.Status()
```

Comment:
The peer.Status().PendingRequestCount will no longer be reliable and peer.Status().ConnectionStatus can only be Available. The abstract list ensures the latter invariant.